code/src/nuvla/connector/kubernetes.py

```diff
@@ -498,3 +505,178 @@ def extract_vm_ports_mapping(self, vm):
     def extract_vm_state(self, vm):
         pass
 
+    # @should_connect
+    def commission(self, payload):
+        """ Updates the NuvlaEdge resource with the provided payload
+        :param payload: content to be updated in the NuvlaEdge resource
+        """
+        # pass
+        if payload:
+            self.api.operation(self.nuvlabox_resource, "commission", data=payload)
+
+
+class K8sSSHKey(Kubernetes):
+    '''Here'''
+    # def __init__(self, job):
+    def __init__(self, **kwargs):
+
+        self.job = kwargs.get("job")
+        if not self.job.is_in_pull_mode:
+            raise ValueError('This action is only supported by pull mode')
+
+        self.api = kwargs.get("api")
+
+        self.nuvlabox_resource = self.api.get(kwargs.get("nuvlabox_id"))
+
+        path = '/srv/nuvlaedge/shared' # FIXME: needs to be parametrised.
+        super(K8sSSHKey, self).__init__(ca=open(f'{path}/ca.pem',encoding="utf8").read(),
+                                          key=open(f'{path}/key.pem',encoding="utf8").read(),
+                                          cert=open(f'{path}/cert.pem',encoding="utf8").read(),
+                                          endpoint=get_kubernetes_local_endpoint())
+
+        # borrowed from nuvlabox.py
+        self.ne_image_registry = os.getenv('NE_IMAGE_REGISTRY', '')
+        self.ne_image_org = os.getenv('NE_IMAGE_ORGANIZATION', 'sixsq')
+        self.ne_image_repo = os.getenv('NE_IMAGE_REPOSITORY', 'nuvlaedge')
+        self.ne_image_tag = os.getenv('NE_IMAGE_TAG', 'latest')
+        self.ne_image_name = os.getenv('NE_IMAGE_NAME', f'{self.ne_image_org}/{self.ne_image_repo}')
+        self.base_image = f'{self.ne_image_registry}{self.ne_image_name}:{self.ne_image_tag}'
+
+    @should_connect
+    def k8s_ssh_key(self, action, pubkey, user_home):
+        '''Doc string'''
+
+        log.debug('We have CA file %s ', self.ca)
+        log.debug('We have certificate file %s ', self.cert)
+        log.debug('We have key file %s ', self.key)
+        log.debug('We have endpoint %s ', self.endpoint)
+        log.debug('We have user home directory %s ', user_home)
```

Comment:
Maybe we can remove the prefix `We have`

Reply:
OK... mainly for my readability in debugging. Removed.